xarray/tests/test_groupby.py

```diff
@@ -1118,7 +1118,8 @@ def test_groupby_math_nD_group() -> None:
     expected = da.isel(x=slice(30)) - expanded_mean
     expected["labels"] = expected.labels.broadcast_like(expected.labels2d)
     expected["num"] = expected.num.broadcast_like(expected.num2d)
-    expected["num2d_bins"] = (("x", "y"), mean.num2d_bins.data[idxr])
+    # mean.num2d_bins.data is a pandas IntervalArray so needs to be put in `numpy` to allow indexing
+    expected["num2d_bins"] = (("x", "y"), mean.num2d_bins.data.to_numpy()[idxr])
```

Comment:
This is technically backwards-incompatible, but an improvement IMO. Just noting in case someone looks this up in the future.

Before:
```
num2d_bins
mean.num2d_bins
<xarray.DataArray 'num2d_bins' (num2d_bins: 2)> Size: 16B
array([Interval(0, 4, closed='right'), Interval(4, 6, closed='right')],
      dtype=object)
Coordinates:
  * num2d_bins  (num2d_bins) object 16B (0, 4] (4, 6]
```

After:
```
ipdb> mean.num2d_bins
mean.num2d_bins
<xarray.DataArray 'num2d_bins' (num2d_bins: 2)> Size: 16B
array([Interval(0, 4, closed='right'), Interval(4, 6, closed='right')],
      dtype=object)
Coordinates:
  * num2d_bins  (num2d_bins) interval[int64, right] 16B (0, 4] (4, 6]
 ```